### PR TITLE
Implement database

### PR DESCRIPTION
**Pull request by drublic**
First draft for this feature.

### CODE REVIEW

File: lib/database/Database.ts [hunk 1]
```diff
@@ -1,59 +1,103 @@
...
   public async set(key: string, content: string): Promise<void> {
-
     if (this.fromMemory) {
       this.memStorage[key] = content;
       return;
     }
 
-    // TODO: store with sequelize model
+    if (this.model) {
+      this.model.upsert({
```
Comment (elmarx) [inline lib/database/Database.ts:47]: when we initially used upsert(), our postgres version (9.3) did not support UPSERT, thus sequelize emulated upsert via a custom SQL-function which turned out to be very costly. That turned out to be a problem when we tried to import (unthrottled) messages from kafka.
So if this upsert is being executed in a performance-critical path, we should double-check if recent sequelize/postgres fully support upsert.

File: package.json [hunk 2]
```diff
@@ -13,21 +13,25 @@
     "ts:lint": "tslint -c tslint.json '**/*.ts'",
     "ts:build": "rm -rf dist/ && tsc -p tsconfig.json",
     "ts:watch": "yarn ts:build --watch",
+    "db:migrate": "./run-migrations.sh",
```
Comment (elmarx) [inline package.json:16]: as your sequelize-files require a build, I'd suggest to add a pre-db:migrate steps that runs `yarn build`, or put the `yarn build` into the `./run-migrations.sh` — if the intent of this npm-task/shell-script is to provide a one-stop-no-worries command.

File: lib/routes/health.ts [hunk 3]
```diff
@@ -0,0 +1,21 @@
+import { Request, Response, Router } from "express";
+
+export default () => {
+  const healthRoutes: Router = Router();
```
Comment (elmarx) [inline lib/routes/health.ts:4]: nice pattern ;)

Comment (drublic) [reply lib/routes/health.ts:4]: Thanks.

File: package.json (hunk 2 above)
Comment (drublic) [reply package.json:16]: Done.

File: lib/database/Database.ts (hunk 1 above)
Comment (drublic) [reply lib/database/Database.ts:47]: Not really critical.

**Pull request merged**